core/src/HIP/Kokkos_HIP_IsXnack.hpp

```diff
@@ -43,7 +40,11 @@ be modified if a better way is determined to check for HMM support in Linux.
 Checking for CONFIG_HMM was considered, but it was not present on El Capitan,
 so we infer its presence is not necessary.
 */
-bool xnack_enabled();
+
+// Returns true iff we detect HSA_XNACK=1 in the environment.
+bool xnack_environment_enabled();
+// Returns true iff we detect CONFIG_HMM_MIROR=y in /boot/config-$(uname -r).
+bool xnack_hmm_found();
```

Comment:
It seems it's the other way around.
```suggestion
// Returns true iff we detect CONFIG_HMM_MIROR=y in /boot/config-$(uname -r).
bool xnack_environment_enabled();
// Returns true iff we detect HSA_XNACK=1 in the environment.
bool xnack_hmm_found();
```